community/cypher/cypher-compiler-3.2/src/main/scala/org/neo4j/cypher/internal/compiler/v3_2/pipes/ExpandAllPipe.scala

```diff
@@ -34,16 +34,18 @@ case class ExpandAllPipe(source: Pipe,
                         (implicit pipeMonitor: PipeMonitor)
   extends PipeWithSource(source, pipeMonitor) {
 
+  private val relationships: ThreadLocal[Iterator[Relationship] with AutoCloseable] =
+    new ThreadLocal[Iterator[Relationship] with AutoCloseable]
```

Comment:
This out-of-bounds state storage irks me. Could we not just close the iterator when we are done with it? i.e. make the caller responsible for closing the iterator it receives from `internalCreateResults` (and have that method return `Iterator[ExecutioContext] with AutoCloseable`).